util/test/start_test.py

```diff
@@ -327,8 +327,12 @@ def test_directory(test, test_type):
                         are_tests = True
                         break
 
+            run_local_sub_test = os.access(os.path.join(dir, "sub_test"), os.X_OK)
```

Comment:
Arguably you could save some time by putting this os.access call inside whatever conditional you add below. Not a big deal, though.